java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -42,11 +44,152 @@
     NativeDepsLoader.loadNativeDeps();
   }
 
+  protected static class NestedHostColumnVector implements AutoCloseable {
+
+    private HostMemoryBuffer data;
+    private HostMemoryBuffer offsets;
+    private HostMemoryBuffer validity;
```

Comment:
Why can't this just be an offHeapState? 

For that matter, correct me if I am wrong, I was thinking the only reason we have to create this NestedHostColumnVector class is because of refcount. Can we extract a common class between the HostColumnVector and the NestedHostColumnVector? Then the HostColumnVector can just extend it and add refCount.

Reply:
This is similar to Bobby's review comment from earlier 
```
Another nit for follow on work. After refactoring the types being returned we probably could combine the majority of this code with the code from copyToHostNestedHelper so that all this does is call into copyToHostNestedHelper and then has an API to translate the top level HostColumnVector.NestedHostColumnVector into a HostColumnVector. But again lets get this in as is and worry about that later.
```
@rjafri , agree this can be done. I relaized it during the review process and can be done incrementally/next version. Let me know if you are ok with a  follow on. Thanks!

Reply:
Thank you @kuhushukla. I think a follow-on should be good

Reply:
https://github.com/rapidsai/cudf/issues/5905 for this issue and the readValue minor comment on consolidation for strings